hazelcast/src/main/java/com/hazelcast/concurrent/atomiclong/AtomicLongService.java

```diff
@@ -176,4 +194,99 @@ public String getQuorumName(String name) {
                 quorumConfigConstructor);
         return quorumName == NULL_OBJECT ? null : (String) quorumName;
     }
+
+    @Override
+    public Runnable prepareMergeRunnable() {
+        IPartitionService partitionService = nodeEngine.getPartitionService();
+        Map<Integer, List<AtomicLongContainer>> containerMap = createHashMap(partitionService.getPartitionCount());
+
+        for (Map.Entry<String, AtomicLongContainer> entry : containers.entrySet()) {
+            AtomicLongContainer container = entry.getValue();
+            if (!(getMergePolicy(container) instanceof DiscardMergePolicy)) {
+                String name = entry.getKey();
+                int partitionId = partitionService.getPartitionId(StringPartitioningStrategy.getPartitionKey(name));
+                if (partitionService.getPartition(partitionId, false).isLocal()) {
```

Comment:
There is also this `IPartitionService.isPartitionOwner()` I think it reads better, but thats just me :)

Reply:
That's a nice one, since it doesn't have this `true|false` parameter (and also doesn't seem to trigger any partition assignment). I'll change all implementation PRs to this, thanks!